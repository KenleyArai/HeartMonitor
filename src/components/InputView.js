import React, { Component } from "react";
import { Text, View } from "react-native";
import PropTypes from "prop-types";
import Button from "./Button";

export default class InputView extends Component<{}> {
  static propTypes = {
    levels: PropTypes.arrayOf(PropTypes.number)
  };

  static defaultProps = {
    levels: [1, 2, 3]
  };

  render() {
    let { levels } = { ...this.props };

    return (
      <View>{levels.map(level => <Button key={level} level={level} />)}</View>
    );
  }
}
