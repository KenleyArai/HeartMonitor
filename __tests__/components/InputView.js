import "react-native";
import React from "react";
import InputView from "../../src/components/InputView";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<InputView />).toJSON();
  expect(tree).toMatchSnapshot();
});
