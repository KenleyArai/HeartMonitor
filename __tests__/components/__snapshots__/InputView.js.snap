// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Button Skeleton
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       
      button
       
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       
      1
       
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Button Skeleton
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       
      button
       
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       
      2
       
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Button Skeleton
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       
      button
       
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       
      3
       
    </Text>
  </View>
</View>
`;
